--- SPECS/subscription-manager.spec.orig	2020-11-24 15:38:49.000000000 +0000
+++ SPECS/subscription-manager.spec	2021-01-16 06:50:51.117915861 +0000
@@ -1,3 +1,5 @@
+%global _name subscription-manager
+
 # Prefer systemd over sysv on Fedora and RHEL 7+
 %global use_systemd 0%{?fedora} || (0%{?rhel} && 0%{?rhel} >= 7) || (0%{?suse_version} && 0%{?suse_version} >= 1315)
 # For optional building of ostree-plugin sub package. Unrelated to systemd
@@ -131,7 +133,7 @@
 
 %global subpackages SUBPACKAGES="%{?include_syspurpose:syspurpose}"
 
-Name: subscription-manager
+Name: subscription-manager-el7
 Version: 1.24.45
 Release: 1%{?dist}
 Summary: Tools and libraries for subscription and repository management
@@ -144,18 +146,18 @@
 # git clone https://github.com/candlepin/subscription-manager.git
 # yum install tito
 # tito build --tag subscription-manager-$VERSION-$RELEASE --tgz
-Source0: %{name}-%{version}.tar.gz
+Source0: %{_name}-%{version}.tar.gz
 # this is a little different from the Source0, because of limitations in tito,
 # namely that tito expects only one source tarball
 %if %{use_cockpit}
-Source1: %{name}-cockpit-%{version}.tar.gz
+Source1: %{_name}-cockpit-%{version}.tar.gz
 %endif
 %if 0%{?suse_version}
 Source2: subscription-manager-rpmlintrc
 %endif
 
 %if 0%{?suse_version} < 1200
-BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
+BuildRoot: %{_tmppath}/%{_name}-%{version}-%{release}-root-%(%{__id_u} -n)
 %endif
 
 # The following macro examples are preceeded by '%' to stop macro expansion
@@ -264,12 +266,24 @@
 BuildRequires: systemd
 %endif
 
+# Remove the Oracle Linux RPMs that obsolete subscription-manager
+Obsoletes: rhn-client-tools
+Obsoletes: rhn-check
+Obsoletes: rhn-setup
+Obsoletes: rhnsd
+Obsoletes: yum-plugin-ulninfo
+Obsoletes: yum-rhn-plugin
+
+# Provide the same things that the subscription-manager RPM provides
+Provides: config(subscription-manager)
+Provides: subscription-manager
+Provides: subscription-manager(%{_arch})
+
 %description
 The Subscription Manager package provides programs and libraries to allow users
 to manage subscriptions and yum repositories from the Red Hat entitlement
 platform.
 
-
 %package -n %{py_package_prefix}-syspurpose
 Summary: A commandline utility for declaring system syspurpose
 Group: System Environment/Base
@@ -282,7 +296,7 @@
 %package -n subscription-manager-plugin-container
 Summary: A plugin for handling container content
 Group: System Environment/Base
-Requires: %{name} = %{version}-%{release}
+Requires: %{_name} = %{version}-%{release}
 
 %description -n subscription-manager-plugin-container
 Enables handling of content of type 'containerImage' in any certificates
@@ -320,15 +334,15 @@
 %package -n subscription-manager-gui
 Summary: A GUI interface to manage Red Hat product subscriptions
 Group: System Environment/Base
-Requires: %{name} = %{version}-%{release}
+Requires: %{_name} = %{version}-%{release}
 Requires: gnome-icon-theme
 
 # We need pygtk3 and gtk2 until rhsm-icon is ported to gtk3
 Requires: rhsm-gtk = %{version}-%{release}
 
 # Renamed from -gnome, so obsolete it properly
-Obsoletes: %{name}-gnome < 1.0.3-1
-Provides: %{name}-gnome = %{version}-%{release}
+Obsoletes: %{_name}-gnome < 1.0.3-1
+Provides: %{_name}-gnome = %{version}-%{release}
 
 %description -n subscription-manager-gui
 This package contains a GTK+ graphical interface for configuring and
@@ -340,7 +354,7 @@
 %package -n subscription-manager-migration
 Summary: Migration scripts for moving to certificate based subscriptions
 Group: System Environment/Base
-Requires: %{name} = %{version}-%{release}
+Requires: %{_name} = %{version}-%{release}
 Requires: rhnlib
 
 # Since the migration data package is not in Fedora, we can only require it
@@ -368,7 +382,7 @@
 %endif
 # See BZ 1581410 - avoid a circular dependency
 %if (0%{?rhel} < 8)
-Requires: %{name} >= %{version}-%{release}
+Requires: %{_name} >= %{version}-%{release}
 %endif
 %if %{with python3}
 Requires: python3-dnf-plugins-core
@@ -429,7 +443,7 @@
 Requires: %{py_package_prefix}-gobject-base
 # plugin needs a slightly newer version of python-iniparse for 'tidy'
 Requires:  %{py_package_prefix}-iniparse >= 0.4
-Requires: %{name} = %{version}-%{release}
+Requires: %{_name} = %{version}-%{release}
 
 %description -n subscription-manager-plugin-ostree
 Enables handling of content of type 'ostree' in any certificates
@@ -523,7 +537,7 @@
 %endif
 
 %prep
-%setup -q
+%setup -n subscription-manager-%{version}
 
 %build
 make -f Makefile VERSION=%{version}-%{release} CFLAGS="%{optflags}" \
@@ -831,7 +845,7 @@
 %config(noreplace) %{_sysconfdir}/dbus-1/system.d/com.redhat.*.conf
 %if %use_systemd
     %attr(644,root,root) %{_unitdir}/*.service
-    %attr(644,root,root) %{_tmpfilesdir}/%{name}.conf
+    %attr(644,root,root) %{_tmpfilesdir}/%{_name}.conf
 %else
     %attr(755,root,root) %{_initrddir}/rhsmcertd
 %endif
@@ -1116,7 +1130,7 @@
 %post -n subscription-manager-gui
 touch --no-create %{_datadir}/icons/hicolor &>/dev/null || :
 %if !0%{?suse_version}
-scrollkeeper-update -q -o %{_datadir}/omf/%{name} || :
+scrollkeeper-update -q -o %{_datadir}/omf/%{_name} || :
 %endif
 %endif
 
