--- subscription-manager.spec.orig	2024-01-24 14:04:19.911554646 +1100
+++ subscription-manager.spec	2024-01-24 14:05:03.102585444 +1100
@@ -589,7 +589,7 @@
 Requires: %{py_package_prefix}-iniparse
 # rpm-python is an old name for python2-rpm but RHEL6 uses the old name
 Requires: %{py_package_prefix}-six
-Requires: subscription-manager-rhsm-certificates
+
 # Required by Fedora packaging guidelines
 %{?python_provide:%python_provide %{py_package_prefix}-rhsm}
 %if %{with python3}
@@ -629,7 +629,7 @@
 Requires: %{py2_package_prefix}-iniparse
 # rpm-python is an old name for python2-rpm but RHEL6 uses the old name
 Requires: %{py2_package_prefix}-six
-Requires: subscription-manager-rhsm-certificates
+
 # Required by Fedora packaging guidelines
 %{?python_provide:%python_provide %{py2_package_prefix}-rhsm}
 Requires: rpm-python
